### DISCUSSION TITLE
Thread safety

### DISCUSSION THREAD

**Discussion by marcovc** [foonathan/memory#160] [Q&A] [answered]
Hi,

I've seen an old closed issue regarding thread safety, but didn't really understood the general case.

If I have two shared_ptrs sharing the same pool allocator, then allocating/deleting them from two different threads is problematic, right?

**Comment by foonathan** [comment, marked as answer]
Yes, it's not thread safe. You can use https://memory.foonathan.net/classfoonathan_1_1memory_1_1thread__safe__allocator.html instead of the pool to make it thread safe.
